### PR TITLE
fix: added an autocommit migration for theme settings.

### PR DESCRIPTION
**Pull request by sondermanish**
Fixes https://github.com/appsmithorg/appsmith/issues/35536

## Automation

/ok-to-test tags="@tag.Git"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/10315694999>
> Commit: f03353bab60f38f3f88bac0303cdc55b5a0edae3
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=10315694999&attempt=1" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.Git`
> Spec:
> <hr>Fri, 09 Aug 2024 08:16:04 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **New Features**
	- Enhanced migration logic for server schema versions, ensuring proper handling of theme settings during migrations from version 6 to 9.
	- Introduced methods to establish and set default theme settings for applications during migration.

- **Bug Fixes**
	- Simplified server version retrieval process, ensuring consistent versioning behavior.

- **Tests**
	- Updated tests to reflect changes in server version expectations, removing unnecessary assertions related to feature flags.
	- Streamlined validation logic in migration tests to focus solely on expected schema versions.

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/JsonSchemaVersions.java [hunk 1]
```diff
@@ -20,8 +18,7 @@ public class JsonSchemaVersions extends JsonSchemaVersionsFallback {
      * @return an Integer which is server version
      */
     @Override
-    @FeatureFlagged(featureFlagName = FeatureFlagEnum.release_git_autocommit_feature_enabled)
     public Integer getServerVersion() {
-        return super.getServerVersion() + 1;
```
Comment (nidhi-nair) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/JsonSchemaVersions.java:25]: Will you be cleaning this up later?

File: app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/MigrationHelperMethods.java [hunk 2]
```diff
@@ -1181,4 +1183,52 @@ private static Set<String> getInvalidDynamicBindingPathsInAction(
...
+        unpublishedApplicationDetail.setThemeSetting(unpublishedThemeSetting);
+    }
+
+    public static void setThemeSettings(Application.ThemeSetting themeSetting) {
+        if (themeSetting.getAppMaxWidth() == null) {
+            themeSetting.setAppMaxWidth(Application.ThemeSetting.AppMaxWidth.LARGE);
+        }
+
+        // since these are primitive values we don't have concept of null, hence putting it to the default of 1.
+        if (themeSetting.getDensity() == 0) {
```
Comment (nidhi-nair) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/MigrationHelperMethods.java:1226]: When would we arrive at this scenario?

File: app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/JsonSchemaVersions.java (hunk 1 above)
Comment (sondermanish) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/JsonSchemaVersions.java:25]: yes, that is when we remove a secondary class with autocommit flags.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/MigrationHelperMethods.java (hunk 2 above)
Comment (sondermanish) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/MigrationHelperMethods.java:1226]: We are not expected to land on this scenario by default, Even if we don't have the density and sizing values not present while deserialising we would start with 1, right? (if not then it's for that).

**Review by nidhi-nair — APPROVED** [review]

**Pull request merged**